src/pages/es/migrate.md

```diff
@@ -0,0 +1,674 @@
+
+```astro
+<script hoist>
+  import './algun-script-externo.js';
+</script>
+```
+
+Nota que Astro empaquetará este script externo con el resto de tu JavaScript del lado del cliente y lo cargará en el contexto de script `type=module`. Es probable que algunos archivos de JavaScript viejos no estén escritos para el contexto `module`, en cuyo caso estos archivos deberían ser actualizados para usar este método.
+
+#### Cómo Resolver Imágenes & Otros Recursos
+
+**1. Path URL Absoluto (Recomendado)**
+
+**Ejemplo:** `<img src="/penguin.png">`
+**Cuándo usar esto:** Si tu recurso se encuentra dentro de la carpeta `public/`.
+
+Si colocas tus imágenes dentro de la carpeta `public/` puedes referenciarlas con un path URL absoluto directo en la plantilla de tu componente. Esta es la forma más simple de referenciar un recurso para ser utilizado y es lo recomendado para la mayoría de los usuarios que recién comienzan a usar Astro.
+
+**2. Importación ESM**
+
+**Ejemplo:** `import imgUrl from './penguin.png'`
+**Cuándo usar esto:** Si tu recurso se encuentra dentro de la carpeta `src/` y quieres aprovechar características de optimización automáticas tales como hashing en nombres de archivos.
+
+Esto funciona dentro de cualquier componente JavaScript o Astro y retorna una URL resuelta a la imagen final. Una vez que tienes la URL resuelta, puedes usarla en cualquier lugar de la plantilla de tu componente.
+
+```astro
+---
+// Ejemplo: Astro incluirá esta imagen en el resultado final
+import imgUrl from './penguin.png';
+---
+<img src={imgUrl} />
+```
+
+De manera similar a cómo Astro maneja CSS, la importación ESM le permite a Astro realizar algunas optimizaciones simples por ti automáticamente. Por ejemplo, cualquier recurso dentro de la carpeta `src/` que sea importando por medio de una importación ESM (ejemplo: `import imgUrl from './penguin.png'`) tendrá su nombre _hasheado_ automáticamente. Esto te permite guardar el archivo en cache en el servidor de manera más agresiva, mejorando la performance para el usuario. En un futuro, Astro agregará más optimizaciones de este estilo.
+
+**Tip:** Si no te gustan las importaciones ESM estáticas, Astro también soporta importaciones ESM dinámicas. Solamente recomendamos esta opción si prefieres este tipo de sintaxis: `<img src={(await import('./penguin.png')).default} />`.
+
+### Deprecado: Procesamiento de `<script>` por Defecto
+
+Anteriormente, todos los elementos `<script>` del HTML generado eran leídos, procesados y empaquetados automáticamente. Este comportamiento ya no es así por defecto. Desde la versión 0.24, debes opt-in al procesamiento de elementos `<script>` por medio del atributo `hoist`. También es requerido `type="module"` para módulos hoisted.
```

Comment:
```suggestion
Anteriormente, todos los elementos `<script>` del HTML generado eran leídos, procesados y empaquetados automáticamente. Este comportamiento ya no es así por defecto. Desde la versión 0.24, debes optar al procesamiento de elementos `<script>` por medio del atributo `hoist`. También es requerido `type="module"` para módulos hoisted.
```

Reply:
Me gusta el cambio, hay otro `opt-in` en la línea 360 pero no se me ocurre bien cómo poner "optar" ahí, alguna sugerencia?
L360:
```
0.24 introdujo una nueva estrategia de *static build* que cambia el comportamiento de algunas características nuevas. Este comportamiento estaba disponible versiones anteriores de Astro por medio de una opt-in flag: `--experimental-static-build`.
```

Reply:
Quedaría mejor llamarlo una `flag opcional`.